src/instruments/src/Common/arinc429.tsx

```diff
@@ -0,0 +1,60 @@
+import { useSimVarValue } from './simVars';
+
+export class Arinc429Word {
+    static SignStatusMatrix = Object.freeze({
+        FailureWarning: 0b00,
+        NoComputedData: 0b01,
+        FunctionalTest: 0b10,
+        NormalOperation: 0b11,
+    });
+
+    static f64View = new Float64Array(1);
+
+    static u32View = new Uint32Array(Arinc429Word.f64View.buffer);
+
+    static f32View = new Float32Array(Arinc429Word.f64View.buffer);
+
+    ssm: number;
```

Comment:
There's a way to make this type fancier and stricter using Typescript:

```typescript
    static SignStatusMatrix = Object.freeze({
        FailureWarning: 0b00,
        NoComputedData: 0b01,
        FunctionalTest: 0b10,
        NormalOperation: 0b11,
    } as const);

    // ...

    ssm: (typeof Arinc429Word.SignStatusMatrix)[keyof typeof Arinc429Word.SignStatusMatrix];
```

To explain what's happening:

First, forget the typeofs, they're just needed because of Typescript's overlaying. Think of it this way:

```typescript
ssm: SignStatusMatrix[keyof SignStatusMatrix]
```

First, we're grabbing the union of keys from SignStatusMatrix:

```typescript
keyof SignStatusMatrix = "FailureWarning" | "NoComputedData" | "FunctionalTest" | "NormalOperation";
```

Now, we index the SignStatusMatrix with those keys:

```typescript
SignStatusMatrix[keyof SignStatusMatrix] = SignStatusMatrix["FailureWarning" | "NoComputedData" | "FunctionalTest" | "NormalOperation"]
```

...which gives us the union of values of SignStatusMatrix:

```typescript
SignStatusMatrix[keyof SignStatusMatrix] = SignStatusMatrix["FailureWarning"] | SignStatusMatrix["NoComputedData"] | SignStatusMatrix["FunctionalTest"] | SignStatusMatrix["NormalOperation"]
```

...which is resolved to:

```typescript
SignStatusMatrix[keyof SignStatusMatrix] = 0 | 1 | 2 | 3;
```

One little detail is that by default, Typescript will widen literals to be helpful. So for example `let foo = true;` will be helpfully widened to `let foo: bool = true;` instead of the "correct" `let foo: true = true`. The same things happens with numbers `let x = 5;` is `let x: number = 5` instead of `let x: 5 = 5`.
So by default, the last code snippet above would actually read:

```typescript
SignStatusMatrix[keyof SignStatusMatrix] = number | number | number | number;
```

...which doesn't help us at all. So to force Typescript to use the literal value, we add the little `as const` marker after your object above:

```typescript
    static SignStatusMatrix = Object.freeze({
        FailureWarning: 0b00,
        NoComputedData: 0b01,
        FunctionalTest: 0b10,
        NormalOperation: 0b11,
    } as const);
```

That way, we correctly get:

```typescript
SignStatusMatrix[keyof SignStatusMatrix] = 0 | 1 | 2 | 3;
```

Of course you could also just define:

```
    ssm: 0 | 1 | 2 | 3;
```

...but where's the fun in that? :-)


In either case, we'll end up where code like this:

```typescript
word.ssm == Arinc429Word.SignStatusMatrix.FailureWarning; // ok
word.ssm == 4;
word.ssm == 3; // ok
word.ssm == "5";
```

will error as expected:

```
test.ts:20:1 - error TS2367: This condition will always return 'false' since the types '0 | 1 | 2 | 3' and '4' have no overlap.

20 word.ssm == 4;
   ~~~~~~~~~~~~~

test.ts:22:1 - error TS2367: This condition will always return 'false' since the types 'number' and 'string' have no overlap.

22 word.ssm == "5";
   ~~~~~~~~~~~~~~~
```

Reply:
A great thorough explanation on increasing type strictness. Thanks for that. I've applied your suggestion and added some error handling of my own.